project_role/models/project_assignment.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ProjectAssignment(models.Model):
+    _name = 'project.assignment'
+    _description = 'Project Assignment'
+
+    name = fields.Char(
+        compute='_compute_name',
+        store=True,
+        index=True,
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        related='project_id.company_id',
+        store=True,
+        readonly=True,
+    )
+    project_id = fields.Many2one(
+        comodel_name='project.project',
+        string='Project',
+        required=True,
+    )
+    role_id = fields.Many2one(
+        comodel_name='project.role',
+        string='Role',
+        required=True,
+        track_visibility='onchange',
```

Comment:
IMO we need to `_inherit = ['mail.thread']` so we can use `track_visibility` attributes.

Reply:
@bodedra as far as I know, it's not required to use email for that, since this changes are tracked in the history that is not emailed by default. Did I get you right?

Reply:
@alexey-pelykh I meant with chatter section to track field.

Reply:
@bodedra got it! I'd appreciate a reference to another module that behaves like you expect 👍

Reply:
@alexey-pelykh Here is reference from _sale_ module. https://github.com/odoo/odoo/blob/12.0/addons/sale/models/sale.py#L22 and https://github.com/odoo/odoo/blob/12.0/addons/sale/views/sale_views.xml#L575

In our case, we need things in .py:

    _inherit = ['mail.thread']

and .xml

    <div class="oe_chatter">
        <field name="message_follower_ids" widget="mail_followers"/>
        <field name="activity_ids" widget="mail_activity"/>
        <field name="message_ids" widget="mail_thread"/>
    </div>

So whenever _Role_ field update, Odoo will update chatter section with old value -> new value.

Reply:
Or You may use https://github.com/bodedra/account-financial-tools/commit/b62c6cacfc4f115888a0cee7ed899258160da021 as a reference which I have recently migrated module.

Reply:
@bodedra thanks for a hint, implemented!